.github/workflows/ci.yml

```diff
@@ -157,117 +104,42 @@ jobs:
-          name: librealm-ios-simulator
-          path: binary/ios
-      - name: Fetch catalyst build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-ios-catalyst
-          path: binary/ios
-
-      - name: Build .xcframework
-        run: |
-          xcodebuild -create-xcframework \
-            -framework ./binary/ios/Release-iphoneos/realm_dart.framework \
-            -framework ./binary/ios/Release-iphonesimulator/realm_dart.framework \
-            -framework ./binary/ios/Release-maccatalyst/realm_dart.framework \
-            -output ./binary/ios/realm_dart.xcframework
-          rm -rf ./binary/ios/Release-*
-
-      - name: Store .xcframework artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: librealm-ios
-          path: binary/ios
-          retention-days: 1
-
-      - name: Delete individual framework artifacts
-        uses: geekyeggo/delete-artifact@v1
-        with:
-          name: |
-            librealm-ios-device
-            librealm-ios-simulator
-            librealm-ios-catalyst
-
   build-android-combined:
     name: Combine Android binaries
-    runs-on: ubuntu-latest
     needs: build-native
+    uses: ./.github/workflows/binary-combine-android.yml
 
-    steps:
-      - name: Fetch x86 build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-x86
-          path: binary/android
-      - name: Fetch x86_64 build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-x86_64
-          path: binary/android
-      - name: Fetch armeabi-v7a build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-armeabi-v7a
-          path: binary/android
-      - name: Fetch arm64-v8a build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-arm64-v8a
-          path: binary/android
-
-      - name: Store combined artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: librealm-android
-          path: binary/android
-          retention-days: 1
-
-      - name: Delete individual build artifacts
-        uses: geekyeggo/delete-artifact@v1
-        with:
-          name: |
-            librealm-android-x86
-            librealm-android-x86_64
-            librealm-android-armeabi-v7a
-            librealm-android-arm64-v8a
+  build-ios-xcframework:
+    name: Combine IOS binaries
+    needs: build-native
+    uses: ./.github/workflows/binary-combine-ios.yml
 
-# Dart jobs
+# Linux jobs
 
-  tests-dart:
-    runs-on: ${{ matrix.platform == 'linux' && 'ubuntu' || matrix.platform }}-${{ matrix.architecture || 'latest' }}
-    name: Tests ${{ matrix.platform }} ${{ matrix.architecture }}
+  dart-tests:
     strategy:
       fail-fast: false
       matrix:
         include:
-          - platform: macos
-          - platform: macos
+          - os: macos
+            app: dm
```

Comment:
nit: Why abbreviate the app names and not name these apps `DartMac` `DartMacArm` `DartWindows` etc, hope the short name does not haunt us in the future while debugging something.

Reply:
ok I see, well in that case we don't have a choice